mex/common/orcid/transform.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any
+
+from pydantic import ValidationError
+
+from mex.common.models import (
+    ExtractedPerson,
+)
+from mex.common.orcid.models.person import FamilyName, GivenNames, OrcidPerson
+from mex.common.primary_source.helpers import get_all_extracted_primary_sources
+
+
+def reduce_metadata(orcid_data: dict[str, Any]) -> dict[str, Any]:
+    """Reduces metadata based on mex assets mapping.
+
+    If the ORCID person's name visibility is "public", returns the email, full name
+    and ORCID ID.
+    If the name is not public, returns only the email and ORCID ID.
+    Returns None dict when orcid person is not exising.
+
+    Args:
+        orcid_data: Retrieved data from orcid.
+
+    Returns:
+        dict: Simplified metadata with email, name and ORCID ID.
+    """
+    if orcid_data == {"result": None, "num-found": 0}:
+        return orcid_data
+    is_public = orcid_data["person"]["name"]["visibility"]
+    return {
+        "orcid-identifier": {
+            "path": orcid_data["orcid-identifier"]["path"],
+        },
+        "person": {
+            "name": {
+                "given-names": orcid_data["person"]["name"]["given-names"]
+                if is_public
+                else None,
+                "family-name": orcid_data["person"]["name"]["family-name"]
+                if is_public
+                else None,
+                "visibility": orcid_data["person"]["name"]["visibility"],
+            },
+            "emails": orcid_data["person"]["emails"],
+        },
+    }
+
+
+def map_to_orcid_person(orcid_data: dict[str, Any]) -> OrcidPerson:
```

Comment:
if the `mex.common.orcid.models` are closer to the raw data we get from orcid, this function would not be needed at all. we can just use `SomePydanticModel.model_validate(data)` instead